analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -450,6 +463,425 @@ public void Finally_NestedFinally()
+
+            validator.TagStates("InInnerTry").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasNoException(x))
+                     .And.ContainSingle(x => HasUnknownException(x));
+
+            validator.TagStates("AfterInnerFinally").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasNoException(x));
+
+            validator.TagStates("AfterOuterFinally").Should().HaveCount(1)
+                     .And.ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void TryCatch_ThrowInTry_SingleCatchBlock()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+    throw new System.Exception();
+    tag = ""UnreachableInFinally"";
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",          // With Exception thrown by Tag("InTry")
+                "InCatch",          // With Exception thrown by `throw`
+                "AfterCatch",
+                "AfterCatch");      // Should go away after https://github.com/SonarSource/sonar-dotnet/pull/5745 is done.
+
+            validator.TagStates("InCatch").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasUnknownException(x))
+                     .And.ContainSingle(x => HasExceptionOfTypeException(x));
+
+            validator.TagStates("AfterCatch").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasUnknownException(x))
+                     .And.ContainSingle(x => HasExceptionOfTypeException(x));
+        }
+
+        [TestMethod]
+        public void TryCatch_ThrowInTry_SingleCatchBlock_ReThrow()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+    throw new System.Exception();
+    tag = ""UnreachableInFinally"";
+}
+catch
+{
+    tag = ""InCatch"";
+    throw;
+}
+tag = ""UnreachableAfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "BeforeTry",
+                "InTry",
+                "InCatch",          // With Exception thrown by Tag("InTry")
+                "InCatch");         // With Exception thrown by `throw`
+
+            validator.TagStates("InCatch").Should().HaveCount(2)
+                     .And.ContainSingle(x => HasUnknownException(x))
+                     .And.ContainSingle(x => HasExceptionOfTypeException(x));
+        }
+
+        [TestMethod]
+        public void TryCatch_ThrowInTry_SingleCatchBlock_ReThrowException()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    Tag(""InTry"");
+    throw new System.Exception();
+    tag = ""UnreachableInFinally"";
+}
+catch (Exception ex)
```

Comment:
Let's add
```
catch (FormatException ex)  // To have a specific exception type
{
    tag = ""InCatch"";
    throw ex;  // To see if the conversion operation works as expected - preparation for the next PR
}
```
and this would also benefit from the `validator.ExitStates` capability mentioned above

Reply:
We already have tests with `catch (ArgumentNullException)` that check the conversion operation. We will add more when we will implement  that.